bundles/org.openhab.binding.etherrain/src/main/java/org/openhab/binding/etherrain/internal/api/EtherRainCommunication.java

```diff
@@ -0,0 +1,248 @@
+                }
+            }
+        }
+
+        return response;
+    }
+
+    public synchronized boolean commandIrrigate(int delay, int zone1, int zone2, int zone3, int zone4, int zone5,
+            int zone6, int zone7, int zone8) {
+        try {
+            sendGet("result.cgi?xi=" + delay + ":" + zone1 + ":" + zone2 + ":" + zone3 + ":" + zone4 + ":" + zone5 + ":"
+                    + zone6 + ":" + zone7 + ":" + zone8);
+        } catch (IOException e) {
+            logger.warn("Could not send irrigate command to etherrain: {}", e.getMessage());
+            return false;
+        }
+
+        return true;
+    }
+
+    public synchronized boolean commandClear() {
+        try {
+            sendGet("/result.cgi?xr");
+        } catch (IOException e) {
+            logger.warn("Could not send clear command to etherrain: {}", e.getMessage());
+            return false;
+        }
+
+        return true;
+    }
+
+    public synchronized boolean commandLogin() throws EtherRainException {
+        try {
+            sendGet("/ergetcfg.cgi?lu=" + ETHERRAIN_USERNAME + "&lp=" + password);
+        } catch (IOException e) {
+            logger.warn("Could not send clear command to etherrain: {}", e.getMessage());
+            return false;
+        }
+
+        return true;
+    }
+
+    public synchronized boolean commandLogout() {
+        try {
+            sendGet("/ergetcfg.cgi?m=o");
+        } catch (IOException e) {
+            logger.warn("Could not send logout command to etherrain: {}", e.getMessage());
+            return false;
+        }
+
+        return true;
+    }
+
+    private synchronized List<String> sendGet(String command) throws IOException {
+        String url = "http://" + address + ":" + port + "/" + command;
+
+        ContentResponse response;
+
+        try {
+            response = httpClient.newRequest(url).timeout(HTTP_TIMEOUT, TimeUnit.SECONDS).send();
+            if (response.getStatus() != HttpURLConnection.HTTP_OK) {
+                logger.warn("Etherrain return status other than HTTP_OK : {}", response.getStatus());
+                return new LinkedList<String>();
+            }
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            logger.warn("Could not connect to Etherrain with exception: {}", e.getMessage());
+            return new LinkedList<String>();
+        }
+
+        return new BufferedReader(new StringReader(response.getContentAsString())).lines().collect(Collectors.toList());
+    }
+
+    private static List<EtherRainUdpResponse> updBroadcast() {
+
+        LinkedList<EtherRainUdpResponse> rList = new LinkedList<EtherRainUdpResponse>();
+
+        // Find the server using UDP broadcast
+
+        try (DatagramSocket c = new DatagramSocket()) {
+
+            c.setSoTimeout(BROADCAST_TIMEOUT);
+            c.setBroadcast(true);
+
+            byte[] sendData = BROADCAST_DISCOVERY_MESSAGE.getBytes();
```

Comment:
Likewise when converting a string to a byte array you should also specify the Charset.